jpype/_jforward.py

```diff
@@ -90,59 +94,53 @@ def _resolveAll():
 
 # This will be used for morphed types, everything must have a base tree
 #  We need to make sure these are fast lookup so we don't change method resoluton cost
-class _JForwardProto(object):
-    pass
+class _JForwardProto:
+    __slots__ = ('_instance', '_symbol')
 
 # resolved symbol
 
 
 class _JResolved(_JForwardProto):
-    def __getattribute__(self, name):
-        instance = object.__getattribute__(self, '_instance')
-        return type(instance).__getattribute__(instance, name)
+    def __getattr__(self, name):
+        return getattr(self._instance, name)
 
     def __setattr__(self, name, value):
-        instance = object.__getattribute__(self, '_instance')
-        return type(instance).__setattr__(instance, name)
+        return setattr(self._instance, name)
 
     def __str__(self):
-        instance = object.__getattribute__(self, '_instance')
-        return str(instance)
+        return str(self._instance)
 
     def __repr__(self):
-        instance = object.__getattribute__(self, '_instance')
-        return repr(instance)
+        return repr(self._instance)
 
-    def __eq__(self, v):
-        instance = object.__getattribute__(self, '_instance')
-        return instance == v
+    def __eq__(self, other):
+        # We swap the order of the __eq__ here to allow other to influence the
+        # result further (e.g. it may be a _JResolved also)
+        return other == self._instance
```

Comment:
This order swapping assumes equality is symmetric in Python (it isn't always, even though it should be)